tensorflow/python/ops/array_ops.py

```diff
@@ -1409,8 +1409,12 @@ def transpose(a, perm=None, name="transpose", conjugate=False):
         gen_array_ops.conjugate_transpose
         if (conjugate and a.dtype.is_complex) else gen_array_ops.transpose)
     if perm is None:
-      rank = gen_array_ops.rank(a)
-      perm = (rank - 1) - gen_math_ops._range(0, rank, 1)
+      if not a.get_shape().ndims:
```

Comment:
This call to get_shape can fail when a is a numpy array. Can you call convert_to_tensor on a?

Reply:
Sorry about that, since it's caught by the tests. Should have diagnosed why the CI pipeline crashes on my machine before sending the patch...

Reply:
ping? It looks like the tests need to be triggered again.